curl ssl error to install kubectl in ci-build-image

In the perspective of tracking all the errors that make the tests flaky, here is one that happens from time to time: the curl command used to install kubectl in the Docker image used in CI fails due to an SSL error.

```sh
[2020-06-05T08:39:23.266Z]  ---> Running in a6291fcaeacc
[2020-06-05T08:39:23.266Z] curl: (56) OpenSSL SSL_read: SSL_ERROR_SYSCALL, errno 104

[2020-06-05T08:39:23.266Z] The command '/bin/sh -c curl -fsSLO https://storage.googleapis.com/kubernetes-release/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl 
  && mv kubectl /usr/local/bin/kubectl 
  && chmod +x /usr/local/bin/kubectl' returned a non-zero code: 56

[2020-06-05T08:39:23.266Z] Makefile:56: recipe for target 'ci-build-image' failed
[2020-06-05T08:39:23.266Z] make[1]: *** [ci-build-image] Error 56
```

https://devops-ci.elastic.co/blue/organizations/jenkins/cloud-on-k8s-e2e-tests-stack-versions/detail/cloud-on-k8s-e2e-tests-stack-versions/108/pipeline/47

As noted [here](https://github.com/curl/curl/issues/4409#issuecomment-555715213), these SYSCALL errors are almost invariably due to server or network issues.

--- a/.ci/Makefile
+++ b/.ci/Makefile
@@ -55,7 +55,7 @@ ci-build-image: DOCKER_PASSWORD = $(shell VAULT_TOKEN=$(VAULT_TOKEN) $(vault) re
 ci-build-image: write-ci-docker-creds
 	@ docker pull -q $(CI_IMAGE) | grep -v -E 'Downloading|Extracting|Verifying|complete' || \
 	( \
-		docker build -f $(ROOT_DIR)/.ci/Dockerfile -t $(CI_IMAGE) --label "commit.hash=$(shell git rev-parse --short --verify HEAD)" $(ROOT_DIR) && \
+		../hack/retry.sh 5 docker build -f $(ROOT_DIR)/.ci/Dockerfile -t $(CI_IMAGE) --label "commit.hash=$(shell git rev-parse --short --verify HEAD)" $(ROOT_DIR) && \
 		../hack/docker-push.sh $(CI_IMAGE) \
 	)
 

--- a/.ci/Makefile
+++ b/.ci/Makefile
@@ -55,7 +55,11 @@ ci-build-image: DOCKER_PASSWORD = $(shell VAULT_TOKEN=$(VAULT_TOKEN) $(vault) re
 ci-build-image: write-ci-docker-creds
 	@ docker pull -q $(CI_IMAGE) | grep -v -E 'Downloading|Extracting|Verifying|complete' || \
 	( \
-		../hack/retry.sh 5 docker build -f $(ROOT_DIR)/.ci/Dockerfile -t $(CI_IMAGE) --label "commit.hash=$(shell git rev-parse --short --verify HEAD)" $(ROOT_DIR) && \
+		../hack/retry.sh 5 docker build \
+			-f $(ROOT_DIR)/.ci/Dockerfile \
+			-t $(CI_IMAGE) \
+			--label "commit.hash=$(shell git rev-parse --short --verify HEAD)" \
+			$(ROOT_DIR) && \
 		../hack/docker-push.sh $(CI_IMAGE) \
 	)
 

